src/inmanta/loader.py

```diff
@@ -140,16 +140,39 @@ def register_code(self, type_name: str, instance: object) -> None:
         if file_name in self.__type_file[type_name]:
             return
 
-        # don't just store this file, but all plugin files in its Inmanta module to allow for importing helper modules
-        all_plugin_files: list[SourceInfo] = list(SourceInfo(file_name, instance.__module__).get_siblings())
+        # get the module
+        def get_inmanta_module_name(instance: object) -> str:
+            module_name = instance.__module__
+            module_parts = module_name.split(".")
+            if module_parts[0] != const.PLUGINS_PACKAGE:
+                raise Exception(
+                    "All instances from which the source is loaded, should be defined in the inmanta plugins package. "
+                    "%s does not match" % module_name
+                )
+            return module_parts[1]
```

Comment:
couldn't we add this to utils so that we can start reusing it? this logic seems to be present in a lot of different places.

Reply:
I factored out the two I could find